resources/channel.sub.js

```diff
@@ -0,0 +1,1003 @@
+                case "undefined":
+                case "null":
+                    break;
+                case "string":
+                case "boolean":
+                    value = item;
+                    break;
+                case "number":
+                    if (item !== item) {
+                        value = "NaN";
+                    } else if (item === 0 && 1/item == Number.NEGATIVE_INFINITY) {
+                        value = "-0";
+                    } else if (item === Number.POSITIVE_INFINITY) {
+                        value = "+Infinity";
+                    } else if (item === Number.NEGATIVE_INFINITY) {
+                        value = "-Infinity";
+                    } else {
+                        value = item;
+                    }
+                    break;
+                case "bigint":
+                case "function":
+                    value = obj.toString();
+                    break;
+                case "remoteobject":
+                    value = {
+                        type: obj.type,
+                        objectId: obj.objectId
+                    };
+                    break;
+                case "sendchannel":
+                    value = item.uuid;
+                    break;
+                case "regexp":
+                    value = {
+                        pattern:item.source,
+                        flags: item.flags
+                    };
+                    break;
+                case "date":
+                    value = Date.prototype.toJSON.call(item);
+                    break;
+                case "error":
+                    value = {
+                        type: item.constructor.name,
+                        name: item.name,
+                        message: item.message,
+                        lineNumber: item.lineNumber,
+                        columnNumber: item.columnNumber,
+                        fileName: item.fileName,
+                        stack: item.stack,
+                    };
+                    break;
+                case "array":
+                case "set":
+                    value = [];
+                    newTarget = {type: "array", value};
+                    for (let child of item) {
+                        stack.push({item: child, target: newTarget});
+                    }
+                    break;
+                case "object":
+                    value = {};
+                    newTarget = {type: "object", value};
+                    for (let [targetKey, child] of Object.entries(item)) {
+                        stack.push({item: child, target: newTarget, targetKey});
+                    }
+                    break;
+                case "map":
+                    value = [];
+                    newTarget = {type: "map", value};
+                    for (let [targetKey, child] of item.entries()) {
+                        stack.push({item: targetKey, target: newTarget, targetKey: "key"});
```

Comment:
The use of `targetKey` as a property here makes for confusion elsewhere I think. Neither "key" nor "values" are keys of anything, and it's also unrelated to the `targetKey` variable here.